src/coreclr/src/jit/lsra.cpp

```diff
@@ -5227,20 +5179,33 @@ void LinearScan::dumpRefPositions(const char* str)
 }
 #endif // DEBUG
 
-bool LinearScan::registerIsFree(regNumber regNum, RegisterType regType)
+//------------------------------------------------------------------------
+// LinearScan::makeRegisterInactive: Make the interval currently assigned to
+//                                   a register inactive.
+//
+// Arguments:
+//    physRegRecord - the RegRecord for the register
+//
+// Return Value:
+//    None.
+//
+// Notes:
+//    It may be that the RegRecord has already been freed, e.g. due to a kill,
+//    or it may be that the register was a copyReg, so is not the assigned register
+//    of the Interval currently occupying the register, in which case this method has no effect.
+//
+void LinearScan::makeRegisterInactive(RegRecord* physRegRecord)
 {
-    RegRecord* physRegRecord = getRegisterRecord(regNum);
-
-    bool isFree = physRegRecord->isFree();
-
-#ifdef TARGET_ARM
-    if (isFree && regType == TYP_DOUBLE)
+    Interval* assignedInterval = physRegRecord->assignedInterval;
+    // It may have already been freed by a "Kill"
+    if ((assignedInterval != nullptr) && (assignedInterval->physReg == physRegRecord->regNum))
     {
-        isFree = getSecondHalfRegRec(physRegRecord)->isFree();
+        assignedInterval->isActive = false;
+        if (assignedInterval->isConstant)
+        {
+            clearNextIntervalRef(physRegRecord->regNum, assignedInterval->registerType);
```

Comment:
Why do we have to do `clearNextIntervalRef()` when marking register inactive and why it is done only for constant intervals?

Reply:
Generally, when we make an interval inactive, we still want to remember its next reference location. However, for a constant interval, even though we set it as inactive, it doesn't actually have a next reference, so we want to clear it.